torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -84,6 +86,13 @@ def register(fn):
         return fn
     return register
 
+def register_rule(call_target):
```

Comment:
The fact that this is named `register_rule` implies it's some sort of superset of the other `register_*_rule` functions. is that the case? Otherwise, should probably name this something else